src/User.cpp

```diff
@@ -852,6 +852,94 @@ bool CUser::IsHostAllowed(const CString& sHostMask) const {
     for (const CString& sHost : m_ssAllowedHosts) {
         if (sHostMask.WildCmp(sHost)) {
             return true;
+        } else {
+            // CIDR notation checker, e.g. "192.0.2.0/24" or "2001:db8::/32"
+
+            // Try to split the string into an IP and routing prefix
+            VCString vsSplitCIDR;
+            const int iSplitCIDRCount = sHost.Split("/", vsSplitCIDR, false);
+            const int iRoutingPrefix = vsSplitCIDR.back().ToInt();
+
+            if (iSplitCIDRCount != 2 || iRoutingPrefix < 0) continue;
+
+            // Convert each IP from a numeric string to an addrinfo
+            addrinfo aiHints;
+            memset(&aiHints, 0, sizeof(addrinfo));
+            aiHints.ai_flags = AI_NUMERICHOST;
+
+            addrinfo* aiHost;
+            int iIsHostValid =
+                getaddrinfo(sHostMask.c_str(), NULL, &aiHints, &aiHost);
+            if (iIsHostValid != 0) continue;
+
+            aiHints.ai_family = aiHost->ai_family;  // Host and range must be in
+                                                    // the same address family
+
+            addrinfo* aiRange;
+            int iIsRangeValid = getaddrinfo(vsSplitCIDR.front().c_str(), NULL,
+                                            &aiHints, &aiRange);
+            if (iIsRangeValid != 0) {
+                freeaddrinfo(aiHost);
+                continue;
+            }
+
+            // "/0" allows all IPv[4|6] addresses
```

Comment:
> 0.0.0.0/0 doesn't allow ::1 because they're different address families

Ah.

> I decided not to match "0.0.0.0" with "127.0.0.1" or "::0" with "::1". Should I implement this matching?

Does such match make sense?